runtime/src/enclave_rpc/client.rs

```diff
@@ -365,18 +353,18 @@ impl RpcClient {
         // Create the command channel.
         let (tx, rx) = mpsc::channel(CMDQ_BACKLOG);
 
-        Self {
-            cmdq: tx.clone(),
-            has_controller: AtomicBool::new(false),
-            controller: Mutex::new(Some(Controller {
-                max_retries: 3,
-                nodes,
-                session: MultiplexedSession::new(builder),
-                transport,
-                cmdq: rx,
-                cmdq_tx: tx,
-            })),
-        }
+        // Create the controller task and start it.
+        let controller = Controller {
+            max_retries: 3,
+            nodes,
+            session: MultiplexedSession::new(builder),
+            transport,
+            cmdq: rx,
+            cmdq_tx: tx.clone(),
+        };
+        tokio::spawn(controller.run());
```

Comment:
Looks good, although I'm not a big fan of creating tasks or goroutines in the constructor, as they are hidden from the caller, not under caller's control and they usually cause troubles in tests (in this case, one must ensure Tokio runtime is available).

Reply:
True, but this way it avoids a needless atomic + mutex.